### PR TITLE
Add theming to footer

### PR DESCRIPTION
**Pull request by Gabss405**
Add theming to ZopaFooter to be able to:
- show/hide `LogoBlock`, `LegalBlock` and `SocialBlock`
- stretch `LegalBlock` full width 
- add optional `legalAmendment` prop to amend `LegalBlock`

### CODE REVIEW

File: src/components/molecules/ZopaFooter/styles/LegalBlock.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import styled from 'styled-components';
 import FlexCol from '../../../layout/FlexCol/FlexCol';
 import { grid } from '../../../../constants';
 
-export const LegalBlock = styled(FlexCol).attrs({ xs: 12, l: 5 })`
```
Comment (admmasters) [inline src/components/molecules/ZopaFooter/styles/LegalBlock.tsx:5]: Is this change required?

Comment (Gabss405) [reply src/components/molecules/ZopaFooter/styles/LegalBlock.tsx:5]: It is to allow LegalBlock to be set full width dynamically based on `isLegalBlockFullWidth`, 
different syntax but I am passing down those attributes [here](https://github.com/zopaUK/react-components/pull/791/files#diff-9b06c978b14544c07234a9db58800fa6f1720e8f13620086fdcedea21d3288f7R152)

**Review by admmasters — APPROVED** [review]
